### ISSUE TITLE
Output valid JSON Schema

### ISSUE THREAD

**Issue by Nijikokun** [Nijikokun/generate-schema#3]
Migrating topic discussion from #2 

- [x] Create a method for generating json schema output.
  
  http://json-schema.org/
  http://en.wikipedia.org/wiki/JSON#JSON_Schema
- [x] Update repl to support both generic, and json schema

**Comment by Nijikokun** [comment]
JSON Schema example input:

```js
console.log(JSON.stringify(GenerateJson({
    "id": 1,
    "name": "A green door",
    "price": 12.50,
    "tags": ["home", "green"]
}), null, 2))
```

Output:

```js
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "type": "object",
  "properties": {
    "id": {
      "type": "number"
    },
    "name": {
      "type": "string"
    },
    "price": {
      "type": "number"
    },
    "tags": {
      "type": "array",
      "items": {
        "type": "string"
      }
    }
  }
}
```

**Comment by Nijikokun** [comment]
Advanced Example Input:

```js
console.log(JSON.stringify(GenerateJson('Product', [
    {
        "id": 2,
        "name": "An ice sculpture",
        "price": 12.50,
        "tags": ["cold", "ice"],
        "dimensions": {
            "length": 7.0,
            "width": 12.0,
            "height": 9.5
        },
        "warehouseLocation": {
            "latitude": -78.75,
            "longitude": 20.4
        }
    },
    {
        "id": 3,
        "name": "A blue mouse",
        "price": 25.50,
        "dimensions": {
            "length": 3.1,
            "width": 1.0,
            "height": 1.0
        },
        "warehouseLocation": {
            "latitude": 54.4,
            "longitude": -32.7
        }
    }
]), null, 2))
```

Output:

```json
{
  "$schema": "http://json-schema.org/draft-04/schema#",
  "title": "Product Set",
  "type": "array",
  "items": {
    "type": "object",
    "properties": {
      "id": {
        "type": "number"
      },
      "name": {
        "type": "string"
      },
      "price": {
        "type": "number"
      },
      "tags": {
        "type": "array",
        "items": {
          "type": "string"
        }
      },
      "dimensions": {
        "type": "object",
        "properties": {
          "length": {
            "type": "number"
          },
          "width": {
            "type": "number"
          },
          "height": {
            "type": "number"
          }
        }
      },
      "warehouseLocation": {
        "type": "object",
        "properties": {
          "latitude": {
            "type": "number"
          },
          "longitude": {
            "type": "number"
          }
        }
      }
    },
    "required": [
      "id",
      "name",
      "price",
      "dimensions",
      "warehouseLocation"
    ],
    "title": "Product"
  }
}
```

**Comment by zaggino** [comment]
You should generate `additionalProperties` key before the `properties` which says `true/false` whether the extra properties not specified by this schema are allowed.

There's a similar keyword `additionalItems` for arrays.

**Comment by zaggino** [comment]
Btw I don't see any unit tests in your repository, so I strongly suggest working on those if you're serious with the library. I personally tend to avoid any third-party stuff which doesn't provide any tests.

**Comment by Nijikokun** [comment]
It doesn't appear that `additionalProperties` required. http://spacetelescope.github.io/understanding-json-schema/reference/object.html

> Btw I don't see any unit tests in your repository, so I strongly suggest working on those if you're serious with the library. I personally tend to avoid any third-party stuff which doesn't provide any tests.

They are on their way, this was something I made to satiate my own personal needs and I shared it with the world. I understand your stance, but that would be a separate issue, please lets stay on topic.

**Comment by Nijikokun** [comment]
Released with `2.0.0`
